src/main/java/com/google/devtools/build/lib/bazel/repository/ArFunction.java

```diff
@@ -0,0 +1,79 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.devtools.build.lib.bazel.repository;
+
+import com.google.common.io.ByteStreams;
+import com.google.devtools.build.lib.bazel.repository.DecompressorValue.Decompressor;
+import com.google.devtools.build.lib.vfs.Path;
+
+import java.io.*;
+import java.util.Date;
+import org.apache.commons.compress.archivers.ar.ArArchiveEntry;
+import org.apache.commons.compress.archivers.ar.ArArchiveInputStream;
+
+/**
+ * Opens a .ar archive file. It ignores the prefix setting because these archives cannot contain
+ * directories.
+ */
+public class ArFunction implements Decompressor {
+
+    public static final Decompressor INSTANCE = new ArFunction();
+
+    // This is the same value as picked for .tar files, which appears to have worked well.
+    private static final int BUFFER_SIZE = 32 * 1024;
+
+    private InputStream getDecompressorStream(DecompressorDescriptor descriptor)
+            throws IOException {
+        return new BufferedInputStream(
+                new FileInputStream(descriptor.archivePath().getPathFile()), BUFFER_SIZE);
+    };
+
+    @Override
+    public Path decompress(DecompressorDescriptor descriptor)
+            throws InterruptedException, IOException {
+        if (Thread.interrupted()) {
+            throw new InterruptedException();
+        }
+
+        try (InputStream decompressorStream = getDecompressorStream(descriptor)) {
+            ArArchiveInputStream arStream = new ArArchiveInputStream(decompressorStream);
+            ArArchiveEntry entry;
+            while ((entry = arStream.getNextArEntry()) != null) {
+                Path filePath = descriptor.repositoryPath().getRelative(entry.getName());
+                filePath.getParentDirectory().createDirectoryAndParents();
+                if (entry.isDirectory()) {
+                    // ar archives don't contain any directory information, so this should never
+                    // happen
+                    continue;
+                } else {
+                    // We do not have to worry about symlinks in .ar files - it's not supported
+                    // by the .ar file format.
+                    try (OutputStream out = filePath.getOutputStream()) {
+                        ByteStreams.copy(arStream, out);
+                    }
+                    filePath.chmod(entry.getMode());
+
+                    Date lastModified = entry.getLastModifiedDate();
```

Comment:
Getting this error message from internal presubmit check:
"Date has a bad API that leads to bugs; prefer java.time.Instant or LocalDate."